backend/controllers/principle-controller.js

```diff
@@ -12,6 +12,14 @@ const passwordRegExp =
   /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,20}$/;
 const emailRegExp = /^\S+@\S+\.\S+$/;
 
+let fetch;
+
+async function loadFetch() {
+  if (!fetch) {
+    fetch = (await import('node-fetch')).default;
```

Comment:
You can import `node-fetch` at the beginning of the file, and use where you need to make a call. That way you don't need a `loadFetch` function and `fetch` variable